### PR TITLE
S8#2

### PR DESCRIPTION
**Pull request by DevPH-Wistkey**

### CODE REVIEW

**Comment by tralkan** [comment]
Hi @DevPH-Wistkey, please update this PR by fetching the latest tyroncoop:s8 - thanks!

**Comment by DevPH-Wistkey** [comment]
Already merged the latest update from tyroncoop/s8, please let me know if there's something I missed. Thanks! @tralkan

**Comment by DevPH-Wistkey** [comment]
Also, I put persist currency task on this PR because this PR is related to balance. But we'll work with one PR per task for upcoming tasks.

**Review by tralkan — CHANGES_REQUESTED** [review]
Some comments pending.

File: src/hooks/isController.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+function controller() {
+  const isController_ = useStore($isController);
+  const Router = useRouter();
+
+  const isController = () => {
+    const path = window.location.pathname.toLowerCase();
+    const username = path.split("/")[1];
+    if (!isController_) {
+      Router.push(`/${username}`);
+      setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (DevPH-Wistkey) [reply src/hooks/isController.ts:15]: Hi @tralkan yes we need it, because it will cause a glitch when the page redirects and shows toast at the same time.

**Review by tralkan — APPROVED** [review]

**Pull request merged**